src/Client.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Http\Client;
+
+/**
+ * Interface for both HTTP Client type
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface Client extends PsrHttpClient, HttpClient
```

Comment:
I'm not sure we are on the same track. The HTTP method interface contains all the conventional HTTP methods for convenience purposes. Of course it is an optional interface, but in some cases it is easier to do `$client->get('/path');` than using a message factory (this is what you call a request factory I guess) and creating requests that way. This is "for the public". An API client will probably construct the requests internally, while in a spaghetti scenario you probably don't want to do that.

IMO the message factory SHOULD NOT care about what the input really is (like is it a multipart form or not?). The message factory and the PsrHttpClient interface assumes that you have prepared data for input. In case of calling a simple get method this is not the case. This is why there were `$data` and `$files` parameters.

With the change I proposed in #42 the special case, when you want to send files in a POST request is still possible, but requires extra efforts from the user. If you want to send string or a StreamInterface, just pass the `$body` parameter to the `post` method. If you want files and other extra sauce for your spaghetti, you have to do a little more than that. But still possible.

Currently I implemented a few Body variations in the body branch, but I am thinking about removing them and leaving only the interface in the adapter/client package. Body implementations can go into a separate package, if needed.